src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -75,7 +75,7 @@ private PatternSyntax ParseDisjunctivePattern(Precedence precedence, bool afterI
             {
                 var orToken = ConvertToKeyword(this.EatToken());
                 var right = ParseConjunctivePattern(precedence, afterIs, whenIsKeyword);
```

Comment:
Not related to this PR, but I noticed we're parsing and-patterns with right-associativity. I suspect that's not by-design.